src/main/java/com/example/bookshop/dto/user/UserLoginRequestDto.java

```diff
@@ -0,0 +1,17 @@
+package com.example.bookshop.dto.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Pattern;
+import jakarta.validation.constraints.Size;
+
+public record UserLoginRequestDto(@NotBlank
+                                  @Size(min = 8, max = 20)
+                                  @Email

```

Comment:
the default regex in `@Email` isn't suitable

you need to specify your own, hence you can specify size within regex and remove `@Size`